rust/numaflow-core/src/source/user_defined.rs

```diff
@@ -253,19 +305,22 @@ mod tests {
 
         let client = SourceClient::new(create_rpc_channel(sock_file).await.unwrap());
 
-        let mut source = Source::new(client)
+        let (mut source, mut lagreader) = new_source(client, 5, 1000)
             .await
```

Comment:
this is confusing, method name is new_source but its creating a source as well as pendingReader